src/sketch.js

```diff
@@ -31,111 +34,219 @@
  * 𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧𖡼.𖤣𖥧
  */
 
-let panda; // selene
-let bunny; // lucy
-let jellyfish; // julie
-let frog; // yen
-/** 𖡼.𖤣𖥧𖡼.𖤣𖥧 STEP 1: ADD CREATURE ABOVE THIS COMMENT 𖡼.𖤣𖥧𖡼.𖤣𖥧 */
-let activeCreature;
-const CREATURES = new Map();
-let CREATURE_SIZE = 100;
-const ARROW_SIZE = CREATURE_SIZE / 4;
-const SMALL_SCREEN_BREAKPOINT = 500;
-let blink = 0; // for arrow blinking
-let touchTarget = null; // for touchscreens
+/**
+ * stores creature and canvas data for a given user sesh
+ * @param {p5.Image[]} creatures - an array of p5.Images
+ */
+class CreatureSession {
+  #unscaledCreatureSize;
+  #blink = 0; // for blinking animations
 
-function preload() {
-  panda = loadImage("/assets/panda.svg");
-  bunny = loadImage("/assets/bunny.svg");
-  jellyfish = loadImage("/assets/jellyfish.svg");
-  frog = loadImage("/assets/frog.svg");
-  /** 𖡼.𖤣𖥧𖡼.𖤣𖥧 STEP 2: ADD CREATURE ABOVE THIS COMMENT 𖡼.𖤣𖥧𖡼.𖤣𖥧 */
-  arrow = loadImage("/assets/arrow.svg");
-}
+  constructor(creatures, arrow, creatureSize = 100, screenBreakpoint = 650) {
+    this.creatures = creatures;
+    this.arrow = arrow;
+    this.#unscaledCreatureSize = creatureSize;
+    this.screenBreakpoint = screenBreakpoint;
+    this.touchTarget = null; // for touchscreens
+  }
 
-function setup() {
-  // smallAxis is the size of the smaller axis if the screen is small, else null
-  const smallAxis =
-    windowWidth < SMALL_SCREEN_BREAKPOINT ||
-    windowHeight < SMALL_SCREEN_BREAKPOINT
-      ? Math.min(windowWidth, windowHeight)
-      : null;
+  #newCanvas() {
+    if (this.smallAxis) {
+      // mobile screens
+      // using screen.* improves the canvas on ios for some reason
+      // iphones also don't rotate screen values in landscape mode
+      // idk if this screws it up on android i should ask someone to test
+      // TODO ask the discord to test on android
```

Comment:
hiii @whgdk could you also try opening the vercel preview for this branch on your pixel? and test it in both portrait and landscape

heres the current link for convenience https://caffeine-web-git-creature-and-833930-caffeine-gardens-projects.vercel.app/